packages/analytics-browser/test/storage/local-storage.test.ts

```diff
@@ -12,4 +13,23 @@ describe('local-storage', () => {
     const localStorage = new LocalStorage();
     expect(await localStorage.isEnabled()).toBe(false);
   });
+
+  test('should drop events when set more than 1000 events and use console log', async () => {
+    const localStorage = new LocalStorage();
+    const errorMock = jest.spyOn(console, 'error');
+
+    await localStorage.set('storage-key', new Array(1001).fill(1));
+    expect(errorMock).toHaveBeenCalledTimes(1);
+    expect(errorMock).toHaveBeenCalledWith('Dropped 1 events because the queue length exceeded 1000.');
+  });
+
+  test('should drop events when set more than 1000 events and use custom logger', async () => {
+    const logger = new Logger();
+    const localStorage = new LocalStorage({ logger });
+    const errorMock = jest.spyOn(logger, 'error');
+
+    await localStorage.set('storage-key', new Array(1001).fill(1));
+    expect(errorMock).toHaveBeenCalledTimes(1);
+    expect(errorMock).toHaveBeenCalledWith('Dropped 1 events because the queue length exceeded 1000.');
+  });
```

Comment:
```suggestion
  test('should drop events when set more than 1000 events and use custom logger', async () => {
    const logger = new Logger();
    const localStorage = new LocalStorage<number[]>({ loggerProvider });
    const errorMock = jest.spyOn(logger, 'error');

    await localStorage.set('storage-key', new Array(1001).fill(1));
    const value = await localStorage.get('storage-key');
    expect(value.length).toBe(1000);
    expect(errorMock).toHaveBeenCalledTimes(1);
    expect(errorMock).toHaveBeenCalledWith('Dropped 1 events because the queue length exceeded 1000.');
  });
```

can we also assert on the actual persisted values?

Reply:
added it